lib/resque.rb

```diff
@@ -271,11 +271,23 @@ def push(queue, item)
     data_store.push_to_queue(queue,encode(item))
   end
 
-  # Pops a job off a queue. Queue name should be a string.
+  # Pops a job off a set of queues. Queue names should be a string.
   #
   # Returns a Ruby object.
-  def pop(queue)
-    decode(data_store.pop_from_queue(queue))
+  def pop(*queues)
+    _, decoded_job = pop_with_queue(*queues)
+
+    decoded_job
+  end
+
+  # Pops a job off a set of queues. Queue names should be strings.
+  #
+  # Returns an array of [queue, Ruby object] or nil.
+  def pop_with_queue(*queues)
```

Comment:
question: Do we pass  the `queues` list ordered from resquepool here?

Reply:
Yes, keeping the queue list order is critical.